src/ngraph/ops/deconvolution.cc

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#include <vector>
+
+#include "ops/deconvolution.h"
+#include "../../operator/nn/deconvolution-inl.h"
+
+#include "ngraph_emitter.h"
+#include "ngraph_emitter_utils.h"
+#include "ngraph_utils.h"
+
+namespace ngraph_bridge {
+
+NgraphNodePtr create_deconvolution(const NgraphNodePtr& data,
+                                   const NgraphNodePtr& filter,
+                                   const ngraph::Shape& out_shape,
+                                   const nnvm::NodePtr& orig_node) {
+  const auto& param =
+      nnvm::get<mxnet::op::DeconvolutionParam>(orig_node->attrs.parsed);
+  const auto data_shape = data->get_shape();
+  const auto filter_shape = filter->get_shape();
+
+  auto n = data_shape.size() - 2;
+  ngraph::CoordinateDiff pad(param.pad.begin(), param.pad.end());
+  ngraph::Strides stride(param.stride.begin(), param.stride.end());
+  ngraph::Strides dilate(param.dilate.begin(), param.dilate.end());
+  auto num_group = param.num_group;
+  if (pad.size() == 0) {
+    pad = ngraph::CoordinateDiff(n, 0);
+  }
+  if (stride.size() == 0) {
+    stride = ngraph::Strides(n, 1);
+  }
+  if (dilate.size() == 0) {
+    dilate = ngraph::Strides(n, 1);
+  }
+
+  NgraphNodePtr conv;
+  if (num_group == 1) {
+    conv = std::make_shared<ngraph::op::ConvolutionBackpropData>(
+        out_shape, filter, data, stride, dilate, pad, pad,
+        ngraph::Strides(n, 1));
+  } else {
+    std::vector<NgraphNodePtr> convolutions(num_group);
```

Comment:
Not sure if this is a real concern or not, but could it make sense to verify the following?
``` c++
assert((  data_shape[1] % num_group) == 0);
assert((filter_shape[0] % num_group) == 0);
```

Reply:
Probably, but it would need to be a check and throw, not an assert. Not sure it really matters, this will only happen if there's an error somewhere coming into the bridge.